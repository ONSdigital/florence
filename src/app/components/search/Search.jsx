import React, { useEffect } from "react";
import { useDispatch } from "react-redux";
import { filterCollections, saveSearch } from "../../config/actions";
import { useInput } from "../../hooks/useInput";
import Magnifier from "../../icons/Magnifier";

const Search = () => {
    const [search, setSearch] = useInput("");
    const dispatch = useDispatch();

    const handleSubmit = e => {
        e.preventDefault();
        dispatch(saveSearch(search.value));
    };

    const handleRest = () => {
        setSearch("");
        dispatch(saveSearch(""));
    };

    return (
        <form className="search__form--inline padding--bottom--1" onSubmit={handleSubmit}>
            <div className="search__input-group">
                <Magnifier classes="search__icon-magnifier" viewBox="0 0 28 28" />
                <input id="search_input" placeholder="Search for a collection name" {...search} />
                {search.value && (
                    <button type="reset" onClick={handleRest} className="btn__close">
                        &times;
                    </button>
                )}
            </div>
            <button className="btn btn--positive" type="submit">
                Search
            </button>
        </form>
    );
};

export default Search;
