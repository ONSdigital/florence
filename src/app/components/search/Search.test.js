import React from "react";
import { WrapperComponent } from "../../../tests/test-utils";
import Search from "./Search.jsx";
import { mount, shallow } from "enzyme";
import renderer from "react-test-renderer";

describe("Search", () => {
    it("matches the snapshot", () => {
        const wrapper = renderer.create(
            <WrapperComponent>
                <Search />
            </WrapperComponent>
        );
        expect(wrapper.toJSON()).toMatchSnapshot();
    });

    it("updates value on change", () => {
        const wrapper = mount(
            <WrapperComponent>
                <Search />
            </WrapperComponent>
        );
        const event = { target: { value: "test" } };
        const onChange = jest.fn();
        wrapper.find("input#search_input").simulate("change", event);
        expect(wrapper.find("input#search_input").getElement().props.value).toBe("test");
    });
});
