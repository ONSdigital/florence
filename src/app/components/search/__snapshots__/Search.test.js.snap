// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search matches the snapshot 1`] = `
<form
  className="search__form--inline padding--bottom--1"
  onSubmit={[Function]}
>
  <div
    className="search__input-group"
  >
    <svg
      className="search__icon-magnifier"
      role="presentation"
      viewBox="0 0 28 28"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="m20.011 16.352h-1.2647l-0.4482-0.4014c1.5689-1.6947 2.5134-3.8948 2.5134-6.2881 0-5.3368-4.6587-9.6627-10.406-9.6627-5.7472 0-10.406 4.3259-10.406 9.6627 0 5.3367 4.6587 9.6626 10.406 9.6626 2.5775 0 4.9469-0.877 6.7719-2.3339l0.4323 0.4163v1.1743l8.0045 7.418 2.3854-2.215-7.9886-7.4328zm-9.6055 0c-3.9862 0-7.2041-2.988-7.2041-6.6895s3.2178-6.6895 7.2041-6.6895c3.9863 0 7.2042 2.988 7.2042 6.6895s-3.2179 6.6895-7.2042 6.6895z"
        fill="#808080"
      />
    </svg>
    <input
      id="search_input"
      onChange={[Function]}
      placeholder="Search for a collection name"
      value=""
    />
    
  </div>
  <button
    className="btn btn--positive"
    type="submit"
  >
    Search
  </button>
</form>
`;
